pkg/controller/cincinnati/cincinnati_controller.go

```diff
@@ -183,6 +188,72 @@ func handleErr(reqLogger logr.Logger, status *cv1alpha1.CincinnatiStatus, reason
 	reqLogger.Error(e, reason)
 }
 
+func (r *ReconcileCincinnati) ensureAdditionalTrustedCA(ctx context.Context, reqLogger logr.Logger, instance *cv1alpha1.Cincinnati, resources *kubeResources) error {
+	defaultOpenShiftConfigNS := "openshift-config"
+
+	// Check if the Cluster is aware of a registry requiring an
+	// AdditionalTrustedCA
+	regconf := &apicfgv1.RegistrySources{}
+	image := &apicfgv1.Image{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: apicfgv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:       defaults.ImageConfigName,
+			UID:        types.UID(utilrand.String(5)),
+			Generation: 1,
+		},
+		Spec: apicfgv1.ImageSpec{
+			RegistrySources: *regconf,
+		},
+	}
+	key, err := client.ObjectKeyFromObject(image)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get object key for ImageConfig")
+	}
+
+	err = r.client.Get(ctx, key, image)
+	if err != nil {
+		return err
+	}
+
+	if image.Spec.AdditionalTrustedCA.Name == "" {
+		return nil
+	}
+
+	// Search for the ConfigMap in openshift-config
+	sourceCM := &corev1.ConfigMap{}
+	err = r.client.Get(ctx, types.NamespacedName{Name: image.Spec.AdditionalTrustedCA.Name, Namespace: defaultOpenShiftConfigNS}, sourceCM)
+	if err != nil && errors.IsNotFound(err) {
+		reqLogger.Info("Found image.config.openshift.io.Spec.AdditionalTrustedCA.Name but did not find expected ConfigMap", "Name", image.Spec.AdditionalTrustedCA.Name, "Namespace", defaultOpenShiftConfigNS)
+		return err
+	} else if err != nil {
+		return err
+	}
+
+	localCM := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
is there a reason you can't skip the TypeMeta?

Reply:
this is some leftover frustration, along with a few other spots.